src/import/ImportWallet.tsx

```diff
@@ -80,8 +78,11 @@ function ImportWallet({ navigation, route }: Props) {
       headerLeft: () => (
         <TopBarTextButtonOnboarding
           title={t('cancel')}
-          // Note: redux state reset is handled by UseBackToWelcomeScreen
-          onPress={() => navigate(Screens.Welcome)}
+          onPress={() => {
+            dispatch(cancelCreateOrRestoreAccount())
+            ValoraAnalytics.track(OnboardingEvents.restore_account_cancel)
+            navigate(Screens.Welcome)
```

Comment:
We need to make sure these actions are triggered when using the Android back button.
(swipe to go back is already prevented on iOS on this screen).

We can use this:

https://github.com/valora-inc/wallet/blob/2b28b691640c52bdeeb1b49a7751c5bdf3b764fe/src/utils/useBackHandler.ts#L4

Reply:
added! 

actually as i was adding this i was thinking, do we need any special navigation at all in this case? the only way that a user can navigate back is by the "cancel" button or with the android hardware back button. in both those places, we have control to clear the navigation stack and go straight back to the Welcome screen. so do we need to worry about clearing the navigation stack before navigating to (or, before this change, when we land on) this screen....?

Reply:
i might be missing something, but the only benefit i see for having something like `navigateClearingOnboardingStack` is that you're able to navigate back to the onboarding carousels from the Welcome screen. i honestly don't think this is valuable functionality though.

Reply:
Yes I think you're right, we can remove clearing the stack now that iOS "swipe to go back" is disabled on this screen.
This would make this whole thing simpler.

The initial intention was to use `UseBackToWelcomeScreen` on most onboarding steps, without having to remember to disabled iOS's  back gesture or the Android back handler, but with the different iterations it ended up being used only on the `ImportWallet` screen.